### PR TITLE
scsynth: CoreAudio: don't complain about mismatched sample rates if numInputBusChannels == 0

### PR DESCRIPTION
**Pull request by snappizz**
Fix for #2608. The scsynth CoreAudio driver checks to make sure that the sample rates of the input and output devices are the same, and if they are not, an error message is produced and scsynth refuses to boot. This commit skips the check if the number of input bus channels is 0. This allows the command-line option "-i 0" to serve as a simple workaround if the user would prefer to simply ignore the input device.

The driver still queries some information about the input device, so this is sort of a bandage. Also, I'm committing blind, so please test.

### CODE REVIEW

File: server/scsynth/SC_CoreAudio.cpp [hunk 1]
```diff
@@ -541,22 +541,22 @@ std::vector<AudioValueRange> GetAvailableNominalSampleRates(const AudioDeviceID&
...
 	AudioObjectPropertyAddress addr = {
 		kAudioDevicePropertyAvailableNominalSampleRates,
 		kAudioObjectPropertyScopeGlobal,
 		kAudioObjectPropertyElementMaster
 	};
-	
+
 	err = AudioObjectGetPropertyDataSize(device, &addr, 0, NULL, &size);
-	
+
```
Comment (vivid-synth) [inline server/scsynth/SC_CoreAudio.cpp]: maybe s/output/output rate/ for clarity?

File: server/scsynth/SC_CoreAudio.cpp [hunk 2]
```diff
@@ -933,8 +933,21 @@ bool SC_CoreAudioDriver::DriverSetup(int* outNumSamplesPerCallback, double* outS
...
+		if (
+			// We do not support mismatched input and output sample rates, but there's no harm in skipping
+			// this check if numInputBusChannels == 0. This allows the user to disable input entirely as
+			// a workaround for a sample rate mismatch.
+			mWorld->mNumInputs > 0 &&
+			!mExplicitSampleRate &&
+			inputStreamDesc.mSampleRate != outputStreamDesc.mSampleRate
+		) {
+			scprintf(
+				"ERROR: Input sample rate is %g, but output is %g. "
```
Comment (vivid-synth) [inline server/scsynth/SC_CoreAudio.cpp]: maybe s/output/output rate/ for clarity?

File: server/scsynth/SC_CoreAudio.cpp [hunk 3]
```diff
@@ -933,8 +933,21 @@ bool SC_CoreAudioDriver::DriverSetup(int* outNumSamplesPerCallback, double* outS
...
+				"Mismatched sample rates are not supported. "
```
Comment (vivid-synth) [inline server/scsynth/SC_CoreAudio.cpp]: What's SC's style regarding adjacent strings for concatenation?

**Comment by vivid-synth** [comment]
Other than my little stylistic questions, looks good to me (although I'm another person who can't test -- just go off @samaaron 's word that it works correctly)

File: server/scsynth/SC_CoreAudio.cpp (hunk 3 above)
Comment (snappizz) [inline server/scsynth/SC_CoreAudio.cpp]: I've seen it in other places in our codebase, but I'm too lazy to check where.

Comment (brianlheim) [inline server/scsynth/SC_CoreAudio.cpp]: C++ will concatenate these just fine. SC does the same thing, no added spaces.

**Pull request merged**